### PR TITLE
method to withdraw excess funds as discussed 

### PR DESCRIPTION
**Pull request by livingrockrises**
(plus extend test for 2nd tx: temp)

before and after

gas used VPM V2  162357. fundingId a tx 1 (unaccounted overhead 35500)  
gas used VPM V2  184592. fundingId a tx 1 (unaccounted overhead 57735)

# Summary

Related Issue: # (issue number)

## Change Type

- [ ] Bug Fix
- [ ] Refactor
- [ ] New Feature
- [ ] Breaking Change
- [ ] Documentation Update
- [ ] Performance Improvement
- [ ] Other

# Checklist

- [ ] My code follows this project's style guidelines
- [ ] I've reviewed my own code
- [ ] I've added comments for any hard-to-understand areas
- [ ] I've updated the documentation if necessary
- [ ] My changes generate no new warnings
- [ ] I've added tests that prove my fix is effective or my feature works
- [ ] All unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published

# Additional Information

# Branch Naming

### CODE REVIEW

**Review by AmanRaj1608 — COMMENTED** [review]
Also need to resolve conflicts

File: contracts/sponsorship/SponsorshipPaymaster.sol [hunk 1]
```diff
@@ -269,6 +279,10 @@ contract SponsorshipPaymaster is
             unaccountedEPGasOverhead *
             effectiveGasPrice;
 
+        // Note: required to calculate excess funds
+        // Review if it should be refunded back to the users and how
+        totalDeductables += balToDeduct;
```
Comment (ankurdubey521) [inline contracts/sponsorship/SponsorshipPaymaster.sol:284]: when is this value reset?

**Comment by ankurdubey521** [comment]
Some quick questions:
1. What are these excess funds?
2. Why are they accruing

File: contracts/sponsorship/SponsorshipPaymaster.sol [hunk 2]
```diff
@@ -184,6 +186,14 @@ contract SponsorshipPaymaster is
         emit GasWithdrawn(msg.sender, withdrawAddress, amount);
     }
 
+    function withdrawExcessFunds(address payable withdrawAddress) public nonReentrant onlyOwner {
```
[the remark this answers is not in the document]
Comment (livingrockrises) [reply contracts/sponsorship/SponsorshipPaymaster.sol:189]: doing now.

File: contracts/sponsorship/SponsorshipPaymaster.sol (hunk 1 above)
Comment (livingrockrises) [reply contracts/sponsorship/SponsorshipPaymaster.sol:284]: it's not reset ever.

**Comment by livingrockrises** [comment]
> Some quick questions:
> 
> 1. What are these excess funds?
> 2. Why are they accruing?

https://www.loom.com/share/3199476cd56e421fbf9b2cb874f21d93?sid=d5c754ba-fc94-4622-9ffb-46d7746687d1

File: contracts/sponsorship/SponsorshipPaymaster.sol (hunk 2 above)
[the remark this answers is not in the document]
Comment (livingrockrises) [reply contracts/sponsorship/SponsorshipPaymaster.sol:197]: done

**Review by ankurdubey521 — APPROVED** [review]

**Pull request closed without merging**